evalml/tests/component_tests/test_stacked_ensemble_regressor.py

```diff
@@ -4,191 +4,232 @@
-    y_pred = clf.predict(X)
+    pl = make_stacked_pipeline()
+    pl.fit(X, y)
+    y_pred = pl.predict(X)
     assert len(y_pred) == len(y)
     assert isinstance(y_pred, pd.Series)
     assert not np.isnan(y_pred).all()
 
-    clf = SklearnStackedEnsembleRegressor(
-        input_pipelines=input_pipelines,
-        final_estimator=RandomForestRegressor(),
-        n_jobs=1,
-    )
-    clf.fit(X, y)
-    y_pred = clf.predict(X)
+    pl = make_stacked_pipeline()
+    pl.component_graph[-1].final_estimator = RandomForestRegressor()
+
+    pl.fit(X, y)
+    y_pred = pl.predict(X)
     assert len(y_pred) == len(y)
     assert isinstance(y_pred, pd.Series)
     assert not np.isnan(y_pred).all()
 
 
-@patch("evalml.pipelines.components.ensemble.SklearnStackedEnsembleRegressor.fit")
-def test_stacked_feature_importance(mock_fit, X_y_regression, stackable_regressors):
+@patch("evalml.pipelines.components.ensemble.StackedEnsembleRegressor.fit")
+def test_stacked_feature_importance(mock_fit, X_y_regression):
     X, y = X_y_regression
-    input_pipelines = [
-        RegressionPipeline([regressor]) for regressor in stackable_regressors
-    ]
-    clf = SklearnStackedEnsembleRegressor(input_pipelines=input_pipelines, n_jobs=1)
+    clf = StackedEnsembleRegressor(n_jobs=1)
     clf.fit(X, y)
     mock_fit.assert_called()
     clf._is_fitted = True
     with pytest.raises(
-        NotImplementedError, match="feature_importance is not implemented"
+        NotImplementedError,
+        match="feature_importance is not implemented for StackedEnsembleClassifier and StackedEnsembleRegressor",
     ):
         clf.feature_importance
+
+
+def test_stacked_same_model_family():
+    graph_en = {
+        "Imputer": ["Imputer", "X", "y"],
+        "Target Imputer": ["Target Imputer", "X", "y"],
+        "OneHot_RandomForest": ["One Hot Encoder", "Imputer.x", "Target Imputer.y"],
+        "OneHot_ElasticNet": ["One Hot Encoder", "Imputer.x", "y"],
+        "Random Forest": ["Random Forest Regressor", "OneHot_RandomForest.x", "y"],
+        "Elastic Net": ["Elastic Net Regressor", "OneHot_ElasticNet.x", "y"],
+        "EN": [
+            "Elastic Net Regressor",
+            "Random Forest.x",
+            "Elastic Net.x",
+            "y",
+        ],
+    }
+
+    graph_rf = {
+        "Imputer": ["Imputer", "X", "y"],
+        "Target Imputer": ["Target Imputer", "X", "y"],
+        "OneHot_RandomForest": ["One Hot Encoder", "Imputer.x", "Target Imputer.y"],
+        "OneHot_ElasticNet": ["One Hot Encoder", "Imputer.x", "y"],
+        "Random Forest": ["Random Forest Regressor", "OneHot_RandomForest.x", "y"],
+        "Elastic Net": ["Elastic Net Regressor", "OneHot_ElasticNet.x", "y"],
+        "Linear Regressor": [
+            "Linear Regressor",
+            "Random Forest.x",
+            "Elastic Net.x",
+            "y",
+        ],
+    }
+
+    RegressionPipeline(component_graph=graph_en)
+    RegressionPipeline(component_graph=graph_rf)
```

Comment:
Unnecessary lines? 😂

Reply:
Also, this might be incorrectly named? :P Since the final estimator is a linear regressor...